### PR TITLE
[Auth] complete auth update & restructure

### PR DESCRIPTION
**Pull request by adam-maj**

### CODE REVIEW

File: packages/wallets/src/evm/interfaces/minimal.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { ethers } from "ethers";
+
+export interface MinimalWallet {
```
Comment (joaquim-verges) [inline packages/wallets/src/evm/interfaces/minimal.ts:3]: Can we call this EVMWallet - and the solana equivalent SolanaWallet ?

File: packages/wallets/src/solana/wallets/base.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { Ecosystem, GenericSignerWallet } from "../../core";
+import { SolanaSigner } from "../interfaces/signer";
+import bs58 from "bs58";
+import nacl from "tweetnacl";
+
+export abstract class AbstractSigner implements GenericSignerWallet {
+  type: Ecosystem = "solana";
+  protected signer: SolanaSigner | undefined;
+
+  public abstract getSigner(): Promise<SolanaSigner>;
```
Comment (joaquim-verges) [inline packages/wallets/src/solana/wallets/base.ts:10]: lets extract this into its own `SolanaWallet` interaface

File: packages/wallets/src/solana/wallets/signer.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { MinimalWallet } from "../interfaces/minimal";
+import { SolanaSigner } from "../interfaces/signer";
+import { AbstractSigner } from "./base";
+
+export class SignerWallet extends AbstractSigner implements MinimalWallet {
```
Comment (joaquim-verges) [inline packages/wallets/src/solana/wallets/signer.ts:5]: oh you have a MinimalWallet here for solana, why not use it in AbstractSigner ?

File: packages/wallets/src/solana/wallets/base.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import { Ecosystem, GenericSignerWallet } from "../../core";
+import { SolanaSigner } from "../interfaces/signer";
+import bs58 from "bs58";
+import nacl from "tweetnacl";
+
+export abstract class AbstractSigner implements GenericSignerWallet {
```
Comment (joaquim-verges) [inline packages/wallets/src/solana/wallets/base.ts:6]: I feel like you're using Signer and Wallet interchangeably and its confusing.

let's use:
- Signer only for things that implement ethers/solana Signer directly and are returned by `wallet.getSigner()`
- Wallet is an object that has a `getSigner()` method, and does wallet things like getAddress(), signMessage() etc

**Review by joaquim-verges — COMMENTED** [review]
@adam-maj i would also rename
AbstractSigner -> AbstractWallet
GenericSignerWallet -> GenericAuthWallet (open to other names)

but need to make it clear cut when we use Signers (privatekey holders) vs Wallets (signer holders) - the hierarchy needs to be clear

**Comment by jnsdls** [comment]
@adam-maj RE the conflicts, almost all of these files now live in `react-core` package (exception is the connect wallet component I believe)

**Comment by adam-maj** [comment]
@joaquim-verges PR should be ready to go and tested everything on express/react. 

Here are the versions: `@thirdweb-dev/auth@3.0.0-dev-60be790` `@thirdweb-dev/react@3.7.0-dev-60be790` `@thirdweb-dev/sdk@3.7.0-dev-60be790` `@thirdweb-dev/wallets@0.2.0-dev-60be790`

Docs: https://docs-bypmstttd.thirdweb-preview.com/auth/server-frameworks/express

Things to test:
- EVM & Solana wallets
- Express & React
- New Callbacks

**Comment by joaquim-verges** [comment]
tested Next w/ EVM and Solana wallets - looking good from my side

**Pull request merged**